ObjC/Encoder+ObjC.mm

```diff
@@ -59,113 +59,101 @@ bool FLEncoder_WriteNSObject(FLEncoder encoder, id obj) FLAPI {
 }
 
 
+
 @implementation NSObject (Fleece)
 - (void) fl_encodeToFLEncoder: (FLEncoder)enc {
-    // Fall back to the internal fl_encodeTo:, which takes a raw C++ Encoder*.
-    [self fl_encodeTo: ((FLEncoderImpl*)enc)->fleeceEncoder.get()];
+    [self fl_encodeToFLEncoder: enc];
 }
 @end
 
 
 @implementation NSObject (FleeceInternal)
-- (void) fl_encodeTo: (Encoder*)enc {
+- (void) fl_encodeToFLEncoder: (FLEncoder)enc {
     // Default implementation -- object doesn't implement Fleece encoding at all.
     NSString* msg = [NSString stringWithFormat: @"Objects of class %@ cannot be encoded",
                      [self class]];
     FleeceException::_throw(EncodeError, "%s", msg.UTF8String);
 }
+
 @end
 
 
 @implementation NSNull (Fleece)
-- (void) fl_encodeTo: (Encoder*)enc {
-    enc->writeNull();
+- (void) fl_encodeToFLEncoder: (FLEncoder)enc {
+    ENCODER_DO(enc, writeNull());
```

Comment:
It's probably better to just call `FLEncoder_WriteNull`, etc. than use the internal `ENCODER_DO` in all these methods.